Assets/Platformer.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+// turn these comments into trello
+// need Effect [i.e. Buffs] and Cooldown Managers #fy 01-15
+// fix player grounding checking to fit the actual player character # fy 01-15
+// give a good pass over scope && lvls of abstraction l8r #fy 01-15
+
+public class Platformer : MonoBehaviour
+{
+    // physical components
+    new Rigidbody2D rigidbody;
+    [SerializeField] Transform _groundBumper;
+
+    // stats
+    [SerializeField] float _baseSpeed;
+    [SerializeField] int _baseJumps;
+    [SerializeField] int _jumpsRemaining;
+    [SerializeField] float _jumpMultiplier;
+    [SerializeField] float _jumpCooldown;
+    [SerializeField] float _leapDistance;
+    [SerializeField] float _distanceToMidair;
+    float _jumpBegan;
+    [SerializeField] float _lastTimeGrounded;
+    [SerializeField] float _groundedForgivenessTime;
+    public bool IsGrounded
+    {
+        get
+        {
+            Collider2D groundPing = Physics2D.OverlapCircle(_groundBumper.position, _distanceToMidair, groundLayer);
+            return groundPing != null ? true : false;
+        }
+    }
+    public bool IsFalling
+    {
+        get
+        {
+            return rigidbody.velocity.y < 0;
+        }
+    }
+    public bool IsRising
+    {
+        get
+        {
+            return rigidbody.velocity.y > 0;
+        }
+    }
+    public bool HasGoodFooting
+    {
+        get
+        {
+            return IsGrounded || Time.time - _lastTimeGrounded <= _groundedForgivenessTime;
+        }
+        set {}
+    }
+    bool _jumpOffCooldown
+    {
+        get
+        {
+            return Time.time - _jumpBegan > _jumpCooldown;
+        }
+        set {}
+    }
+    public bool CanJump
+    {
+        get
+        {
+            return this.HasGoodFooting || (_jumpsRemaining > 0 && _jumpOffCooldown);
```

Comment:
Another little quibble not worth launching into a commit to fix, but I've found that I can sometimes triple jump. something about the timing of one jump right after another leads to the player jumping into the air, but retaining a jumps remaining of 2. I think it's a timing thing, maybe related to HasGoodFooting or _jumpOffCooldown? Anyway, something to be aware of anyway.